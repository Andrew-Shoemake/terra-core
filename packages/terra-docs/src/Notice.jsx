import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import ThemeContext from 'terra-theme-context';

import styles from './Notice.module.scss';

const cx = classNames.bind(styles);

const getTitle = (variant) => {
  if (variant === 'deprecation') {
    return 'Deprecation Notice';
  }

  if (variant === 'maintenance') {
    return 'Component in Maintenance';
  }

  return 'error';
};

const propTypes = {
  /**
  * The main body text of the notice.
  */
  children: PropTypes.string,
  /**
  * The type of notice.
  */
  variant: PropTypes.oneOf(['deprecation', 'maintenance']),
};

const defaultProps = {
  variant: 'deprecation',
};

const Notice = ({
  variant,
  children,
}) => {
  const theme = React.useContext(ThemeContext);
  return (
    <h2 className={cx('notice', variant, theme.className)}>
      {getTitle(variant)}
      <p className={cx('paragraph')}>
        {children}
      </p>
    </h2>
  );
};

Notice.propTypes = propTypes;
Notice.defaultProps = defaultProps;

export default Notice;
