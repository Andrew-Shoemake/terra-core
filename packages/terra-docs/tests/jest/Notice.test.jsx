import React from 'react';
import Notice from '../../src/Notice';

describe('Placeholder', () => {
  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(<Notice />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a deprecation notice with text', () => {
    const wrapper = shallow(
      <Notice variant="deprecation">
        notice text
      </Notice>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a maintenance notice with text', () => {
    const wrapper = shallow(
      <Notice variant="maintenance">
        notice text
      </Notice>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
