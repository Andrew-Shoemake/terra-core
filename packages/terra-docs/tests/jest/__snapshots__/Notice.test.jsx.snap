// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder should render a default component 1`] = `
<h2
  className="notice deprecation"
>
  Deprecation Notice
  <p
    className="paragraph"
  />
</h2>
`;

exports[`Placeholder should render a deprecation notice with text 1`] = `
<h2
  className="notice deprecation"
>
  Deprecation Notice
  <p
    className="paragraph"
  >
    notice text
  </p>
</h2>
`;
